testnet/stacks-node/src/tests/signer/v0.rs

```diff
@@ -2483,6 +2500,233 @@ fn signers_broadcast_signed_blocks() {
+    assert_eq!(signer_slot_ids.len(), num_signers);
+
+    let get_last_block_hash = || {
+        let blocks = test_observer::get_blocks();
+        let last_block = blocks.last().unwrap();
+        let block_hash =
+            hex_bytes(&last_block.get("block_hash").unwrap().as_str().unwrap()[2..]).unwrap();
+        Sha512Trunc256Sum::from_vec(&block_hash).unwrap()
+    };
+
+    let last_block_hash = get_last_block_hash();
+
+    let slot_id = 0_u32;
+
+    let get_last_block_response = |slot_id: u32| {
+        let mut stackerdb = StackerDB::new(
+            &naka_conf.node.rpc_bind,
+            StacksPrivateKey::new(), // We are just reading so don't care what the key is
+            false,
+            reward_cycle,
+            SignerSlotID(0), // We are just reading so again, don't care about index.
+        );
+        let latest_msgs = StackerDB::get_messages(
+            stackerdb
+                .get_session_mut(&MessageSlotID::BlockResponse)
+                .expect("Failed to get BlockResponse stackerdb session"),
+            &[slot_id],
+        )
+        .expect("Failed to get message from stackerdb");
+        let latest_msg = latest_msgs.last().unwrap();
+        let SignerMessage::BlockResponse(BlockResponse::Accepted(accepted)) = latest_msg else {
+            panic!("Latest message from slot #{slot_id} isn't a block acceptance");
+        };
+        accepted.clone()
+    };
+
+    let latest_acceptance = get_last_block_response(slot_id);
+    assert_eq!(latest_acceptance.signer_signature_hash, last_block_hash);
+
+    info!(
+        "---- Last idle timeout: {} ----",
+        latest_acceptance.tenure_extend_timestamp
+    );
+
+    // Now, mine a few nakamoto blocks with just transfers
+
+    let mut sender_nonce = 0;
+
+    let mut last_acceptance = latest_acceptance;
+
+    for i in 0..num_txs {
+        info!("---- Mining interim block {} ----", i + 1);
+        let transfer_tx = make_stacks_transfer(
+            &sender_sk,
+            sender_nonce,
+            send_fee,
+            naka_conf.burnchain.chain_id,
+            &recipient,
+            send_amt,
+        );
+        submit_tx(&http_origin, &transfer_tx);
+        sender_nonce += 1;
+
+        let info_before = get_chain_info(&signer_test.running_nodes.conf);
+        wait_for(30, || {
+            let info = get_chain_info(&signer_test.running_nodes.conf);
+            Ok(info.stacks_tip_height > info_before.stacks_tip_height)
+        })
+        .expect("Timed out waiting for nakamoto block to be mined");
+
+        let latest_acceptance = get_last_block_response(slot_id);
+        let last_block_hash = get_last_block_hash();
+
+        assert_eq!(latest_acceptance.signer_signature_hash, last_block_hash);
+        // Because the block only contains transfers, the idle timeout should not have changed
+        assert_eq!(
+            last_acceptance.tenure_extend_timestamp,
+            latest_acceptance.tenure_extend_timestamp
+        );
```

Comment:
@jferrant this test is for verifying that STX transfers don't effect the `tenure_extend_timestamp` on a block response. Even though this is built on top of #5470 , the timestamp stays the same (so this test passes). I'm a bit confused as to why that's happening, as currently the `calc_tenure_extend_timestamp` just does `block.processed - block.proposed`, without regards for execution cost of the block.

Reply:
Sorry this was before I had made the necessary changes (was just a placeholder). Should rebase and retry :)

Reply:
Will need the miner side in place before this will work.

Reply:
Are you sure? Now that the signer is getting processed time from the node, this test should fail. I haven't actually ran it yet since you last merged recently.

Reply:
Oh woops I thought I was looking at the other test. :) Let me take a look.

Reply:
stx_transfers_dont_effect_idle_timeout does indeed pass with the updated code :D

Reply:
Resolving - the issue is just that blocks were getting too fast and we were losing precision.